src/viewProviders/flinkArtifacts.ts

```diff
@@ -50,12 +39,52 @@ export class FlinkArtifactsViewProvider
             const loader = CCloudResourceLoader.getInstance();
             this._artifacts = await loader.getFlinkArtifacts(this.computePool!);
           } catch (error) {
-            this.logger.error("Failed to load Flink artifacts", { error });
+            let showNotification = false;
+            let message = "Failed to load Flink artifacts.";
+
+            if (isResponseError(error)) {
+              const status = error.response.status;
+              // Only show notifications for error status codes
+              if (status >= 400 && status < 600) {
+                showNotification = true;
+                switch (status) {
+                  case 401:
+                    message = "Authentication required to load Flink artifacts.";
+                    break;
+                  case 403:
+                    message =
+                      "Failed to load Flink artifacts. Please check your permissions and try again.";
+                    break;
+                  case 404:
+                    message = "Flink artifacts not found for this compute pool.";
+                    break;
+                  case 429:
+                    message = "Too many requests. Please try again later.";
+                    break;
+                  case 503:
+                    message =
+                      "Failed to load Flink artifacts. The service is temporarily unavailable. Please try again later.";
+                    break;
+                  default:
+                    message = "Failed to load Flink artifacts due to an unexpected error.";
+                    break;
+                }
+              }
+              logError(error, "Failed to load Flink artifacts");
+            } else if (error instanceof Error) {
+              message =
+                "Failed to load Flink artifacts. Please check your connection and try again.";
+              showNotification = true;
+              logError(error, "Failed to load Flink artifacts");
+            } else {
+              message =
+                "Failed to load Flink artifacts. Please check your connection and try again.";
+              showNotification = true;
+              logError(error, "Failed to load Flink artifacts");
+            }
 
-            // Check if this is an auth error (401 Unauthorized)
-            if (isResponseError(error) && error.response.status === 401) {
-              // Signal that the auth session is invalid
-              ccloudAuthSessionInvalidated.fire();
+            if (showNotification) {
+              await showErrorNotificationWithButtons(message);
```

Comment:
We don't want to `await` this since it'll block and keep the view loading indicator active until the notification is closed or a button is clicked:
```suggestion
              void showErrorNotificationWithButtons(message);
```